apps/server/src/modules/tool/service/validation/school-external-tool-validation.service.spec.ts

```diff
@@ -439,4 +442,408 @@ describe('SchoolExternalToolValidationService', () => {
+			it('should return without error when type matches param value', async () => {
+				const correctTypeParam: CustomParameterDO = customParameterDOFactory.build({
+					name: 'correctType',
+					scope: CustomParameterScope.SCHOOL,
+					type: CustomParameterType.NUMBER,
+				});
+				const { schoolExternalToolDO, schoolExternalToolId } = setup(
+					{
+						parameters: [correctTypeParam],
+					},
+					{
+						parameters: [{ name: correctTypeParam.name, value: '1234' }],
+					}
+				);
+
+				const func = () => service.validateUpdate(schoolExternalToolId, schoolExternalToolDO);
+
+				await expect(func()).resolves.not.toThrowError('tool_param_type_mismatch:');
+			});
+
+			it('should throw exception when type not matches param value', async () => {
+				const wrongTypeParam: CustomParameterDO = customParameterDOFactory.build({
+					name: 'wrongType',
+					scope: CustomParameterScope.SCHOOL,
+					type: CustomParameterType.NUMBER,
+				});
+				const { schoolExternalToolDO, schoolExternalToolId } = setup(
+					{
+						parameters: [wrongTypeParam],
+					},
+					{
+						parameters: [{ name: wrongTypeParam.name, value: '17271hsadas' }],
+					}
+				);
+
+				const func = () => service.validateUpdate(schoolExternalToolId, schoolExternalToolDO);
+
+				await expect(func()).rejects.toThrowError('tool_param_type_mismatch:');
+			});
+		});
+
+		describe('when checking parameter type boolean', () => {
+			it('should return without error when type matches param value', async () => {
+				const correctTypeParam: CustomParameterDO = customParameterDOFactory.build({
+					name: 'correctType',
+					scope: CustomParameterScope.SCHOOL,
+					type: CustomParameterType.BOOLEAN,
+				});
+				const { schoolExternalToolDO, schoolExternalToolId } = setup(
+					{
+						parameters: [correctTypeParam],
+					},
+					{
+						parameters: [{ name: correctTypeParam.name, value: 'true' }],
+					}
+				);
+
+				const func = () => service.validateUpdate(schoolExternalToolId, schoolExternalToolDO);
+
+				await expect(func()).resolves.not.toThrowError('tool_param_type_mismatch:');
+			});
+
+			it('should throw exception when type not matches param value', async () => {
+				const wrongTypeParam: CustomParameterDO = customParameterDOFactory.build({
+					name: 'wrongType',
+					scope: CustomParameterScope.SCHOOL,
+					type: CustomParameterType.BOOLEAN,
+				});
+				const { schoolExternalToolDO, schoolExternalToolId } = setup(
+					{
+						parameters: [wrongTypeParam],
+					},
+					{
+						parameters: [{ name: wrongTypeParam.name, value: '17271hsadas' }],
+					}
+				);
+
+				const func = () => service.validateUpdate(schoolExternalToolId, schoolExternalToolDO);
+
+				await expect(func()).rejects.toThrowError('tool_param_type_mismatch:');
+			});
+		});
+
+		describe('when checking parameter type auto_courseId', () => {
+			it('should return without error when type matches param value', async () => {
```

Comment:
The typeChecker always returns true for that type, so "when type matches param value" is misleading here.